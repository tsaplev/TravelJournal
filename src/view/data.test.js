const { getHtmlData } = require("./data");
const { matchers } = require("jest-json-schema");

expect.extend(matchers);

jest.mock("../models/trips");
jest.mock("../models/flights");

describe("Fetch data from all sources", () => {
  test("collect data for the main template", async () => {
    const data = await getHtmlData();

    expect(data).toMatchSchema({
      type: "object",
      required: [
        "siteTitle",
        "title",
        "subTitle",
        "googleMapApiKey",
        "flightradarUsername",
        "shareData",
        "countries",
        "trips",
      ],
    });
    expect(data.countries).toMatchSnapshot();
    expect(data.trips).toMatchSnapshot();
    expect(data.shareData).toEqual(
      expect.stringMatching(/<script>window\.shareData = {(.*)};<\/script>/)
    );
  });
});
