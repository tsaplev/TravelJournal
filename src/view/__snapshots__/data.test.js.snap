// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch data from all sources collect data for the main template 1`] = `
"
    <ul class=\\"countries-list\\">
      <li class=\\"countries-list__country\\">
        <span class=\\"country\\">ğŸ‡¦ğŸ‡º Australia: </span>Sydney
      </li>
    </ul>
    
    <ul class=\\"countries-list\\">
      <li class=\\"countries-list__country\\">
        <span class=\\"country\\">ğŸ‡·ğŸ‡º Russia: </span>Moscow, Sevastopol
      </li>
    </ul>
    
    <ul class=\\"countries-list\\">
      <li class=\\"countries-list__country\\">
        <span class=\\"country\\">ğŸ‡ºğŸ‡¸ United States of America: </span>New York
      </li>
    </ul>
    "
`;

exports[`Fetch data from all sources collect data for the main template 2`] = `"<p><strong class=\\"year\\">2014</strong><br><span class='log'>ğŸ‡·ğŸ‡º Moscow <span class='time'>January 7 - 9</span></span><br><span class='log'>ğŸ‡·ğŸ‡º Sevastopol <span class='time'>February 20 - March 26</span></span><br></p><p><strong class=\\"year\\">2007</strong><br><span class='log'>ğŸ‡¦ğŸ‡º Sydney <span class='time'>November 5 - 18</span></span><br></p><p><strong class=\\"year\\">1995</strong><br><span class='log'>ğŸ‡ºğŸ‡¸ New York <span class='time'>May 10 - 20</span></span><br></p>"`;
